plugins/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,511 @@
+            beast_http_session(
+                std::shared_ptr<http_plugin_state> plugin_state,
+                asio::io_context* ioc)
+                : ioc_(ioc)
+                , plugin_state_(plugin_state)
+            {
+                plugin_state_->requests_in_flight += 1;
+                req_parser_.body_limit(plugin_state_->max_body_size);
+            }
+
+            virtual ~beast_http_session() {
+                plugin_state_->requests_in_flight -= 1;
+            }
+
+            void
+            do_read()
+            {
+                //fc_ilog( logger, "get_lowest_layer()" ); 
+                // Set the timeout.
+                beast::get_lowest_layer(
+                    derived().stream()).expires_after(std::chrono::seconds(30));
+
+                //fc_ilog( logger, "async_read()" ); 
+                // Read a request
+                http::async_read(
+                    derived().stream(),
+                    buffer_,
+                    req_parser_,
+                    beast::bind_front_handler(
+                        &beast_http_session::on_read,
+                        derived().shared_from_this()));
+            }
+
+            void
+            on_read(
+                beast::error_code ec,
+                std::size_t bytes_transferred)
+            {
+                boost::ignore_unused(bytes_transferred);
+
+                // This means they closed the connection
+                if(ec == http::error::end_of_stream)
+                    return derived().do_eof();
+
+                if(ec && ec != asio::ssl::error::stream_truncated)
+                    return fail(ec, "read");
+
+
+                auto req = req_parser_.get();
+                // Send the response
+                handle_request(std::move(req));
+            }
+
+            void
+            on_write(
+                bool close,
+                beast::error_code ec,
+                std::size_t bytes_transferred)
+            {
+                //fc_ilog( logger, "ignore_unused()" ); 
+                boost::ignore_unused(bytes_transferred);
+
+                if(ec) {
+                    ec_ = ec;
+                    handle_exception();
+                    return fail(ec, "write");
+                }
+
+                if(close)
+                {
+                    // This means we should close the connection, usually because
+                    // the response indicated the "Connection: close" semantic.
+                    return derived().do_eof();
+                }
+
+                // We're done with the response so delete it
+                // res_ = nullptr;
+
+                //fc_ilog( logger, "do_read()" ); 
+
+                // Read another request
+                do_read();
+            }
+
+
+            // TODO 
+            bool allow_host(const http::request<http::string_body>& req) {
+                /*
+                bool is_secure = con->get_uri()->get_secure();
```

Comment:
code could use a pass through to remove commented out code

Reply:
This should now be cleaned up